### PR TITLE
Vanity Domain support, and general swagger updates

### PR DESCRIPTION
**Pull request by sburke-ce**
## Description

This PR extends the client configuration to enable use with Vanity Domains, by adding the following optional configuration variables:
```go
    // Vanity domain type one of "", "tenant" or "server".
   VanityDomainType string `json:"vanity_domain_type"`

    // Tenant id required when VanityDomainType is "tenant" or "server"
    TenantID string

    // Authorization server id required when VanityDomainType is "server".
    ServerID string
```

For backward-compatibility, the behavior is unchanged with `VanityDomainType = ""`. When `Routing VanityDomainType` is:
* "tenant" - TenantID must be specified in the Config
* "server" - TenantID and ServerID must be specified in the Config

I have tested the Vanity Domain support with the Marks & Spencer custom-consent application.

This PR also includes an updated swagger spec, in order to provide access to the Transient OTP APIs in swagger-public.yaml.
I have updated `ace-client-go/specs/` with all of the yaml files in `acp/api/handler/`, _except_ for openbanking.yaml, because it is not obvious how to carry those across.

## Type of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Tests (extending the test suite)
- [ ] Refactor (internal improvement that doesn't change product functionality)
- [ ] Other (if none of the other choices apply)

### CODE REVIEW

File: spec/swagger-developer.yaml [hunk 1]
```diff
@@ -1685,7 +1690,7 @@ host: localhost:8443
 info:
   description: Cloudentity Authorization Control Plane Public API
   title: Cloudentity Authorization Control Plane API
-  version: 1.13.1
+  version: latest
```
Comment (sburke-ce) [inline spec/swagger-developer.yaml]: Oops - need to set a correct version.

**Review by morus12 — APPROVED** [review]

File: spec/swagger-developer.yaml (hunk 1 above)
Comment (sburke-ce) [reply spec/swagger-developer.yaml:1701]: I notice that `version: latest` comes from ACP - should i fix these to a particular release, or is 'latest' OK?

File: client.go [hunk 2]
```diff
@@ -170,8 +170,14 @@ type Config struct {
 	// HttpClient is the client to use. Default will be used if not provided.
 	HttpClient *http.Client `json:"-"`
 
-	// Routing mode
+	// Routing mode, one of "", "tenant" or "server".
 	RoutingMode string `json:"routing_mode"`
```
Comment (sburke-ce) [inline client.go:174]: I think that this config variable might be easier to understand, if the name was tied to vanity domains in some way.
Perhaps `VanityDomainType`, or something similar, would be better?

**Review by mbilski — APPROVED** [review]

File: spec/swagger-developer.yaml (hunk 1 above)
Comment (mbilski) [reply spec/swagger-developer.yaml:1701]: latest is ok

**Review by ikawalec — APPROVED** [review]

File: client.go (hunk 2 above)
Comment (sburke-ce) [reply client.go]: OK, i'm going to rename RoutingMode -> VanityDomainType.

**Pull request merged**